commercial/app/controllers/commercial/ContentApiOffersController.scala

```diff
@@ -111,6 +115,64 @@ class ContentApiOffersController(contentApiClient: ContentApiClient, capiAgent:
     }
   }
 
+
+  case class CapiSingle(articleHeadline: String, articleUrl: String,
+                        articleText: Option[String], articleImage: Seq[ImageElement],
+                        audioTag: Boolean, galleryTag: Boolean,
+                        videoTag: Boolean, branding: Option[Branding])
+
+  object CapiSingle {
+    import ElementsFormat._
+
+    def fromContent(contentType: ContentType): CapiSingle = {
+      val content = contentType.content
+      val branding = BrandHunter.findContentBranding(contentType, Edition.defaultEdition)
+
+      CapiSingle(content.trail.headline, content.metadata.webUrl, content.trail.fields.trailText, content.elements.images, content.tags.isAudio,
+        content.tags.isGallery, content.tags.isVideo, branding)
+    }
+    implicit val writesCapiSingle: Writes[CapiSingle] = Json.writes[CapiSingle]
+  }
+
+  private def renderNative(format: Format, isMulti: Boolean) = Action.async { implicit request =>
+
+    val optKeyword = request.getParameter("k")
+
+    val latestContent = optKeyword.map { keyword =>
+      // getting twice as many, as we filter out content without images
+      lookup.latestContentByKeyword(keyword, 8)
+    }.getOrElse(Future.successful(Nil))
+
+    latestContent onFailure {
+      case NonFatal(e) => log.error(s"Looking up content by keyword failed: ${e.getMessage}")
+    }
+
+    val specificContent: Future[Seq[model.ContentType]] = capiAgent.contentByShortUrls(specificIds)
+
+    specificContent onFailure {
+      case NonFatal(e) => log.error(s"Looking up content by short URL failed: ${e.getMessage}")
+    }
+
+    val futureContents = for {
+      specific <- specificContent
+      latestByKeyword <- latestContent
+    } yield {
+      (specific ++ latestByKeyword.filter(_.trail.trailPicture.nonEmpty)).distinct take 4
+    }
+
+    futureContents.map((content: Seq[model.ContentType]) => {
+      val response = content.head
+      val capiSingle = CapiSingle.fromContent(response)
+      Cached(60.seconds) {
+        JsonComponent(capiSingle)
+      }
+    })
+
+  }
+
+  def nativeJson = renderNative(jsonFormat, isMulti = false)
```

Comment:
`renderNative` and `renderItems` have a lot in common, you could take the data extraction part out of both so that they only take care of the rendering?